File: python/1288_Remove_Covered_Intervals/solution.py
# Slow O(N ** 2) solution

class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        answer = 0
        for i in range(len(intervals)):
            overlap = False
            for j in range(len(intervals)):
                if i == j:
                    continue
                if intervals[i][0] >= intervals[j][0] and intervals[i][1] <= intervals[j][1]:
                    overlap = True
            if not overlap:
                answer += 1
        return answer

File: rust/1288_Remove_Covered_Intervals/solution.rs
// Slow O(N ** 2) solution

impl Solution {
    pub fn remove_covered_intervals(intervals: Vec<Vec<i32>>) -> i32 {
        let mut answer = 0;
        let mut overlap = false;
        for i in 0..intervals.len() {
            overlap = false;
            for j in 0..intervals.len() {
                if i == j {
                    continue;
                }
                else if intervals[i][0] >= intervals[j][0] && intervals[i][1] <= intervals[j][1] {
                    overlap = true;
                }
            }
            if !overlap {
                answer += 1;
            }
        }
        answer
    }
}
